test/testUtil.ts

```diff
@@ -152,3 +163,53 @@ export const createFakeConnection = function (): Connection {
     },
   } as any;
 };
+/**
+ * Validate that a CSV file has X records
+ */
+export async function validateCsv(
+  filePath: string,
+  columnDelimiter: ColumnDelimiterKeys,
+  totalQty: number
+): Promise<void> {
+  const csvReadStream = fs.createReadStream(filePath);
+  let recordCount = 0;
+
+  await pipeline(
+    csvReadStream,
+    new csvParse({ columns: true, delimiter: ColumnDelimiter[columnDelimiter] }),
+    new PassThrough({
+      objectMode: true,
+      transform(_chunk, _encoding, callback) {
+        recordCount++;
+        callback(null, null);
+      },
+    }),
+    // dummy writable
+    new Writable({
+      write(_chunk, _encoding, callback) {
+        callback();
+      },
+    })
+  );
+
+  expect(totalQty).to.equal(recordCount);
+}
+
+/**
+ * Validate that a JSON file has X records
+ */
+export async function validateJson(filePath: string, totalqty: number): Promise<void> {
+  // check records have expected fields
+  const fieldsRes = await exec(
+    `jq 'map(has("Id") and has("Name") and has("Phone") and has("AnnualRevenue")) | all' ${filePath}`,
+    {
+      shell: 'pwsh',
+    }
+  );
+  expect(fieldsRes.stdout.trim()).equal('true');
+
+  // check all records were written
+  const lengthRes = await exec(`jq length ${filePath}`, { shell: 'pwsh' });
+
+  expect(parseInt(lengthRes.stdout.trim(), 10)).equal(totalqty);
+}
```

Comment:
test helper to ensure:
1. a valid JSON file was written (`jq` fails it not)
2. the total number of records processed by the job were written
3. queried fields were written on each record